app/routes.py

```diff
@@ -1 +1,267 @@
+
+    SLACK_TOKEN = os.environ.get("SLACK_API_KEY")
+
+    request_body = {
+        "token": SLACK_TOKEN,
+        "channel": "task-notifications",
+        "text": message }
+
+    requests.post(path, data=request_body)
+    
+
+
+    return make_response(jsonify(response), 200)
+
+
+
+
+"""GOAL ROUTES"""
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    
+    if request_body.get("title") is None:
+        return make_response ({"details": "Invalid data"}, 400)
+        request_body = request.get_json()
+    
+    new_goal = Goal.from_dict(request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response = {"goal": new_goal.to_dict()}
+
+    return make_response(jsonify(response), 201)
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    
+    goals = Goal.query.all()
+
+    goals_list =[goal.to_dict() for goal in goals]
+
+    return jsonify(goals_list), 200
+    
+    
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+
+    goal = validate_model(Goal, goal_id)
+    response = {"goal": goal.to_dict()}
+
+    return make_response(jsonify(response), 200)
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_task(goal_id):
+    goal=validate_model(Goal, goal_id)
+    request_body=request.get_json()
+
+    if request_body.get("title") is None:
+        return make_response(f"Some additional information needed to update goal{goal.id}{goal.title}",400)
+
+    goal.title=request_body["title"]
+
+
+    db.session.commit()
+
+    response = {"goal": goal.to_dict()}
+
+    return make_response(jsonify(response), 200)
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+    response = {"details": (f"Goal {goal.id} \"{goal.title}\" successfully deleted")}
+
+    return make_response(jsonify(response), 200)
+
+
+
+
+
+"""NESTED ROUTES"""
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def create_task_id(goal_id):
+
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    
+    if request_body.get("task_ids") is None:
+        return make_response ({"details": "Invalid data"}, 400)
+    
+    
+    for task_id in request_body["task_ids"]:
+        task = validate_model(Task, task_id)
+        task.goal_id = goal_id 
+        
+    
+    db.session.commit()
+    
+    
+    
+    response = {"id": goal.id, "task_ids": request_body["task_ids"]}
+
+
+
+    return make_response(jsonify(response), 200)
+
+
+
+
+
+
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_task_from_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    response = {"id":goal.id, "title": goal.title, "tasks":[task.to_dict() for task in goal.tasks]}
+
+    return jsonify(response), 200
+
+
+
+
+
+
+
```

Comment:
Quite a lot of space between these later functions, best to leave one or two lines at most